Add Learner Profiles

astroDimitrios opened the pull request:
Closes #4 - adding learner profiles and pathways.

jonaslindemann commented:
I like the concept of learner profiles, but I think it is important that it doesn't complicate the structure of a course. Perhaps have separate document on with suggested paths throuigh the course.

cmcooling commented:
I think Learner Profiles can be useful as a tool for developing content, but I'm less convinced in their utility as a student-facing thing. I worry they could bog the student down. I think that the profiles themselves probably are useful.

One suggestion I had comes from how we organise our own courses and dependencies, using [this tool](https://observablehq.com/embed/@imperialrcds/learning-paths?cells=page,Title,Buttons). Using this users can select topics they're interested in and the tool picks out a learning path. Something similar could allow students to pick things that are relevant to them and the tool could dynamically generate a path through the material. Just a thought that sprang to mind.

astroDimitrios commented:
Thanks @cmcooling @jonaslindemann! I was under the mistaken impression that the Profiles page was only visible in Instructor View - turns out it is the other way around. Perhaps this is something I can ask of the Workbench devs as to why that is. I think a Pathways page should be visible to Learners. I love the pathways tool - is it an Imperial only site or are there ways we could add to / use it?

joeHickson commented on profiles/learner-profiles.md:
```diff
@@ -1,5 +1,22 @@
+
+***Malak is mathematics graduate student working on new timestepping methods.*** Malaks previous work was mostly theoretical although he knows basic Python. Malak needs to use Fortran to prototype his new routines which will interface with other Fortran code. Malak has been told by his supervisor that he should set up his preferred text editor for Fortran development.
+
+***Joe has been employed by a tech company to update their Fortran code.*** Joe has years of experience developing code but has not used Fortran in a production environment. They have been asked to update legacy code to modern standards, including implementing some form of testing and using standard libraries. Joe is not sure if there is a standard code style that they can adopt.
```
Is there value in splitting Joe into Experienced programmer in compiled language vs Experienced programmer in non-compiled language? As a Joe who came to fortran as an experienced programmer in non-compiled softly typed languages it was a double whammy of compiling and strict types that a c++ programmer would not face.

joeHickson replied:
I realise the C++ background variant is covered by Ionas and the python background (potentially at a more junior level) by Tahani

joeHickson commented on profiles/learner-profiles.md:
```diff
@@ -1,5 +1,22 @@
+
+***Ionas, a researcher joining the JULES (Joint UK Land Environment Simulator) partnership.*** They will be adding new functionality to JULES in Fortran. They have extensive experience with C and C++ and have been told there are differences between Fortran and C/C++ that they should be aware of. They are keen to learn Fortran but must fit their learning inbetween teaching duties and other researc ...
+
+***Malak is mathematics graduate student working on new timestepping methods.*** Malaks previous work was mostly theoretical although he knows basic Python. Malak needs to use Fortran to prototype his new routines which will interface with other Fortran code. Malak has been told by his supervisor that he should set up his preferred text editor for Fortran development.
```
I feel like Malak knows their way around in matlab ;)

astroDimitrios commented:
I have made a small update to Joe's profile. I have also split the profiles out and moved them into the instructor notes. This leaves the learning pathways in the learner profiles page which is visible to students (couldn't work out an easy way to change the name as even the page title for learner-profiles.md is not respected). The flowchart for the pathways is a mermaid diagram which will require #7 to render. I have added it here so you can see the rendered draft without checking out the whole branch:

```mermaid
flowchart TD
    accTitle: Learner Pathways for the Introduction to Modern Fortran Course
    accDescr {This diagram shows the possible learning pathways after completing the Introduction to Modern Fortran course.
    The Intermediate and Advanced Topics in Modern Fortran courses are in the pre-alpha stage.
    The MPI and OpenMP courses are from ARCHER2, the UK's national supercomputing facility, courses.}
    A["Introduction to Modern Fortran <br>(this course)"] --> B
    B[Intermediate Modern Fortran] --> C & D & E
    C[Advanced Topics<br>in Modern Fortran]
    D[Message-passing<br>Programming with MPI]:::archer2
    E[Shared Memory<br>Programming with OpenMP]:::archer2
    classDef archer2 stroke:#64c2ab
    click D href "https://www.archer2.ac.uk/training/courses/#message-passing-programming-with-mpi" "Link to the MPI Archer2 Course"
    click E href "https://www.archer2.ac.uk/training/courses/#shared-memory-programming-with-openmp" "Link to the OpenMP Archer2 Course"
```

joeHickson reviewed (approved):
LGTM
